### ISSUE TITLE
Filter_gen interpolation issue

### ISSUE THREAD

**Issue by willend** [McStasMcXtrace/McCode#1149]
Bugeports from Peter and Andreas, TUM:

mail 1/2
```
Dear Peter,

we just stumbled across an unexpected behavior of the Filter_gen component.

We use it to modify our source flux to simulate varying reactor operation conditions. The transmission file is of two column [wavevector multiply] type. The data points are NOT evenly spaced, with a rather high data point density in the thermal, and a low one in the cold neutron range.

If we run our simulation we observe a clear difference between McStas 2.4 and the current McStas 2.7, with likely the 2.7 version having wrong (up to a factor 2) resulting intensities.

Has there been a change in the component or in one of the underlying libs (interpolation, rebinning of the table)?

Best regards,

Peter
```
![verg_version_withoutCS](https://user-images.githubusercontent.com/8593355/131621619-e7b34673-ceda-4bc5-bab2-0f5ddf96c08f.jpg)

mail 2/2
```
Dear Peter,

I had a closer look on the problem using a test case. See the attached files and the figure, which suggests that the table interpolation has a "floor" like feature. This really seamed to appear somewhere between McStas 2.4 and 2.7...

Best regards

Peter

```
![bpngknnlplfbmble](https://user-images.githubusercontent.com/8593355/131621692-28d65603-fb6e-4d43-8d69-2d3165d8a02f.png)

Attachments to last mail:

[box_shape.mul.txt](https://github.com/McStasMcXtrace/McCode/files/7088700/box_shape.mul.txt)
[Test_filter_gen.instr.txt](https://github.com/McStasMcXtrace/McCode/files/7088701/Test_filter_gen.instr.txt)

**Comment by willend** [comment]
This in fact looks more an issue of read_table-lib than the component itself...

**Comment by willend** [comment]
It turns out that this was a specific issue for 2.7:

[2_7_others.pdf](https://github.com/McStasMcXtrace/McCode/files/7089884/2_7_others.pdf)
[2_7_vs_patched.pdf](https://github.com/McStasMcXtrace/McCode/files/7089885/2_7_vs_patched.pdf)


The problem had already been fixed by @ebknudsen on master and 3.0 branches, the solution is to pick a fresh set of read_table-lib files from
https://raw.githubusercontent.com/McStasMcXtrace/McCode/master/common/lib/share/read_table-lib.c
https://raw.githubusercontent.com/McStasMcXtrace/McCode/master/common/lib/share/read_table-lib.h 

or simply await the forthcoming release(s)
